extension/types/data-types.ts

```diff
@@ -0,0 +1,45 @@
+export type DictData = {
```

Comment:
I think we talked about it a bit on discord but after seeing it; I think this types based cateogorization is not as useful as a semantic categorization (were files are grouped based on their semantic purpose rather than on a more general classification)

It's also fairly unclear what constitutes a type for some of the files here.

Finally, we should be careful when adding new exports as that creates public interface contracts that we'll need to be careful in the future.

I think if we want to add some grouping then we could group all the dictionary handling code together (maybe renaming data.ts (probably spliting it in the future since it contains fairly distinct functiaonality around loading and lookups now))

(we could also punt this til later since it's not so complicated yet and it's good to keep code moving in their own PRs (otherwise git doesn't detect it and gives you a bad diff as you can see here)